File: javascript/leetcode/56. Merge Intervals.js
/**
 * @param {number[][]} intervals
 * @return {number[][]}
 */
var merge = function (intervals) {
  const ans = [];

  intervals = intervals.sort((a, b) => a[0] - b[0]);

  intervals.forEach((el) => {
    if (!ans.length || ans[ans.length - 1][1] < el[0]) ans.push(el);
    else ans[ans.length - 1][1] = Math.max(ans[ans.length - 1][1], el[1]);
  });

  return ans;
};

File: c++/leetcode/56. Merge Intervals.cpp
class Solution
{
public:
  vector<vector<int>> merge(vector<vector<int>> &intervals)
  {
    vector<vector<int>> ans;
    sort(intervals.begin(), intervals.end(), [](vector<int> &a, vector<int> &b) {
      return a[0] < b[0];
    });

    for (auto i : intervals)
    {
      if (ans.empty() || ans[ans.size() - 1][1] < i[0])
        ans.push_back(i);
      else
        ans[ans.size() - 1][1] = max(ans[ans.size() - 1][1], i[1]);
    }

    return ans;
  }
};